docs/features/enumerations.rst

```diff
@@ -0,0 +1,143 @@
+
+Using a FieldDict
+-----------------
+
+In a FieldDict, you can specify an Enum for the type. This will generate the corresponding enum in the `.proto` file.
+
+.. code-block:: python
+
+    @grpc_action(
+        request=[{"name": "enum_example", "type": MyEnum}],
+        response=[{"name": "value", "type": "string"}],
+    )
+
+Here is the generated enum in `.proto` file:
+
+.. code-block:: protobuf
+
+    message Request {
+        MyEnum.Enum enum_example = 1;
+    }
+
+    message MyEnum {
+        enum Enum {
+            ENUM_UNSPECIFIED = 0;
+            FIRST = 1;
+            SECOND = 2;
+        }
+    }
+
+Using an Annotated Serializer or Model field
+--------------------------------------------
+
+If you want to use choices in a Serializer or Model, you can use `TextChoices` or `IntegerChoices` and use `Annotated` to specify the choices you use as metadata.
+
+DSG is not able to directly access the choices in the field as they are normalized as a tuple missing the keys; that's why we use `Annotated`.
+
+.. warning::
+
+    If you don't use `Annotated`, no enum will be generated in the `.proto` file; instead, your field will simply be a string or int32 type.
+
+Example in a model:
+
+.. code-block:: python
+
+    from typing import Annotated
+
+    class MyModel(models.Model):
+        class MyEnum(models.TextChoices):
+            FIRST = "FIRST", "First"
+            SECOND = "SECOND", "Second"
+
+        my_field: Annotated[models.CharField, MyEnum] = models.CharField(choices=MyEnum)
+
+Example in a Serializer:
+
+.. code-block:: python
+
+    from typing import Annotated
+
+    class MySerializer(proto_serializers.ProtoSerializer):
+        class MyEnum(models.TextChoices):
+            FIRST = "FIRST", "First"
+            SECOND = "SECOND", "Second"
+
+        my_field: Annotated[serializers.ChoiceField, MyEnum] = serializers.ChoiceField(choices=MyEnum)
+
+Here is the generated enum in `.proto` file:
+
+.. code-block:: protobuf
+
+    message Request {
+        MyEnum.Enum my_field = 1;
+    }
+
+    message MyEnum {
+        enum Enum {
+            ENUM_UNSPECIFIED = 0;
+            FIRST = 1;
+            SECOND = 2;
+        }
+    }
+
+Note that if you use a `ModelProtoSerializer`, and your model has `Annotated` on fields containing choices, you don't have to annotate them again in the serializer.
+
+Adding Comments
+---------------
+
+You can add comments at the enumeration level by adding a Docstring to it, or at the members level by adding Annotated to them.
+
+.. code-block:: python
+
+    from typing import Annotated
+
+    class MyModel(models.Model):
+        class MyEnum(models.TextChoices):
+            """My enum comment"""
+
+            FIRST : Annotated[tuple, ["Comment", "on two lines"]] = "FIRST", "First"
```

Comment:
Yes indeed this is not exactly the same use case as this one is a completely new annotation where protocomment is for differenciate simple string to a proto comment.
What I was wondering is: Is it possible to use the same Annotated logic for other proto comment and not only the only in enum ?

Reply:
So for exemple, you mean the possibility to use an Annotated on a Serializer field to build a ProtoComment instead of having to use help_text ?
If so, i think that it would be a bit too much honestly, plus we already use Annotated on ChoiceField to get the Enum type